subsys/net/l2/wifi/wifi_utils.c

```diff
@@ -24,7 +24,7 @@ LOG_MODULE_REGISTER(net_wifi_utils, CONFIG_NET_L2_WIFI_MGMT_LOG_LEVEL);
 /* Ensure 'strtok_r' is available even with -std=c99. */
 char *strtok_r(char *str, const char *delim, char **saveptr);
 
-static const uint16_t valid_5g_chans_20mhz[] = {32, 36, 40, 44, 48, 52, 56, 60, 64, 68, 96, 100,
+static const uint8_t valid_5g_chans_20mhz[] = {32, 36, 40, 44, 48, 52, 56, 60, 64, 68, 96, 100,
```

Comment:
@krish2718 is this assumption always valid (that channel number is always < 256)?
According to https://en.wikipedia.org/wiki/List_of_WLAN_channels it seems so.

Reply:
Yes, for now unless more channels open up.

Reply:
Yes, that would future proof (though unlikely it will happen).